src/Extensions/UsedOnTableExtension.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace SilverStripe\LinkField\Extensions;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\Core\Extension;
+use SilverStripe\LinkField\Models\FileLink;
+use SilverStripe\ORM\DataObject;
+
+class UsedOnTableExtension extends Extension
+{
+    public function updateUsageAncestorDataObjects(array &$ancestorDataObjects, DataObject $dataObject): void
+    {
+        if (!is_a($dataObject, FileLink::class)) {
+            return;
+        }
+        $owner = $dataObject->Owner();
+        $ancestorDataObjects[] = $owner;
+        if (class_exists(BaseElement::class)
+            && is_a($owner, BaseElement::class)
+            && $owner->hasMethod('getTopPage')
+        ) {
+            $page = $owner->getTopPage();
```

Comment:
```suggestion
        ) {
            $page = $owner->hasExtension(\DNADesign\Elemental\TopPage\DataExtension) ? $owner->getTopPage() : $owner->getPage();
```
Or just use `getPage()` directly.

Also obviously use a use statement for the extension FQCN - probably with an alias that makes it clear what the extension actually is since it's got a stupid name.
```php
use DNADesign\Elemental\TopPage\DataExtension as TopPageExtension;
```

Reply:
Updated